src/viur/core/modules/history.py

```diff
@@ -0,0 +1,742 @@
+                self.client.create_table(
+                    bigquery.Table(
+                        self.tablepath,
+                        schema=self.schema
+                    )
+                )
+                self.table = self.client.get_table(self.tablepath)
+
+        logging.debug(f"found bigquery table {self.table!r}")
+        return self.table
+
+    def write_row(self, data):
+        assert self.client and self.table
+        if res := self.client.insert_rows(self.table, [data]):
+            raise ValueError(res)
+
+    # ---------------------------------------------------------------------------------------------
+    # FIXME: Started below code previously which generates the schema from the ViurHistorySkel
+    # This is for now abandoned, but the code should not be thrown away, it could be useful.
+
+    IGNORE_BONES = (
+        "viurCurrentSeoKeys",
+    )
+
+    # def __init__(self):
+    #     super().__init__()
+    #     self.client = bigquery.Client()
+    #     self.table = None
+    #     self.schema = None
+    #     self.path = f"""{conf.instance.project_id}.history.default"""
+
+    def __generate_schema(self, skel):
+        # FIXME: Currently not in use!
+        assert self.schema is None
+        self.schema = []
+
+        for name, bone in skel.items():
+            if name in self.IGNORE_BONES:
+                continue
+
+            def bone_to_schema(name, bone, descr=None):
+                if isinstance(bone, RelationalBone):
+                    return [
+                        bone_to_schema(
+                            f"{name}_{relname}",
+                            relbone,
+                            descr=f"""{(descr or "") + bone.descr} - {relbone.descr}"""
+                        )
+                        for relname, relbone in bone._refSkelCache().items()
+                    ]
+                elif isinstance(bone, BooleanBone):
+                    datatype = "BOOLEAN"
+                elif isinstance(bone, JsonBone):
+                    datatype = "JSON"
+                elif isinstance(bone, DateBone):
+                    datatype = "DATETIME"
+                elif isinstance(bone, NumericBone):
+                    datatype = "NUMERIC"
+                elif isinstance(bone, RawBone):
+                    datatype = "BYTES"
+                else:
+                    datatype = "STRING"
+
+                return {
+                    "type": datatype,
+                    "name": name,
+                    "mode": "REPEATED" if bone.multiple else "REQUIRED" if bone.required else "NULLABLE",
+                    "description": descr or bone.descr,
+                }
+
+            schema = bone_to_schema(name, bone)
+
+            if isinstance(schema, dict):
+                self.schema.append(schema)
+            else:
+                self.schema.extend(schema)
+
+    def _get_table(self, skel):
+        # FIXME: Currently not in use!
```

Comment:
then please remove this.

This looks identical like `select_or_create_table` anyway